### PR TITLE
Fix unlocking of encrypted boot partition in storage-ng setups

### PR DESCRIPTION
**Pull request by okurz**
Verification runs:
* SLE15 x86_64 cryptlvm: http://lord.arch/tests/425
* SLE12SP4 x86_64 cryptlvm: http://lord.arch/tests/426
* SLE12SP4 x86_64 lvm-full-encrypt: http://lord.arch/tests/430

Related progress issue: https://progress.opensuse.org/issues/27829

### CODE REVIEW

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -336,6 +336,7 @@ sub wait_boot {
             select_console('iucvconn');
         }
         else {
+            workaround_type_encrypted_passphrase if get_var('S390_ZKVM');
```
Comment (asmorodskyi) [inline lib/opensusebasetest.pm]: just to confirm , you currently in branch where variable ARCH not equal s390x . and you check for variable S390_ZKVM, right ?

File: lib/opensusebasetest.pm [hunk 2]
```diff
@@ -374,23 +375,20 @@ sub wait_boot {
             assert_screen "grub2", 15;
         }
         elsif (match_has_tag("migration-source-system-grub2") or match_has_tag('grub2')) {
-            send_key "ret";    # boot to source system
+            # no action required
```
Comment (asmorodskyi) [inline lib/opensusebasetest.pm]: if no action required why you not just drop whole branch ? I would understand if you would have final else at the end but this is not the case

File: lib/opensusebasetest.pm [hunk 3]
```diff
@@ -374,23 +375,20 @@ sub wait_boot {
...
         }
         elsif (get_var("LIVETEST")) {
             # prevent if one day booting livesystem is not the first entry of the boot list
             if (!match_has_tag("boot-live-" . get_var("DESKTOP"))) {
                 send_key_until_needlematch("boot-live-" . get_var("DESKTOP"), 'down', 10, 5);
             }
-            send_key "ret";
```
Comment (asmorodskyi) [inline lib/opensusebasetest.pm]: after deleting this send_key you can move "sub-if" condition up 

```
elsif (get_var("LIVETEST") && !match_has_tag("boot-live-" . get_var("DESKTOP"))) { # prevent if one day booting livesystem is not the first entry of the boot list
send_key_until_needlematch("boot-live-" . get_var("DESKTOP"), 'down', 10, 5);
              }

```

File: lib/opensusebasetest.pm [hunk 4]
```diff
@@ -410,7 +410,7 @@ sub wait_boot {
         select_console('sut');
     }
 
-    # on s390x svirt is encryption unlocked with workaround_type_encrypted_passphrase before this wait_boot
+    # on s390x svirt is encryption unlocked with workaround_type_encrypted_passphrase before here
```
Comment (asmorodskyi) [inline lib/opensusebasetest.pm]: could you please change this to 

> "on s390x svirt  encryption is unlocked with workaround_type_encrypted_passphrase before here"

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (okurz) [reply lib/opensusebasetest.pm]: No, I'm within the if-branch started in L324, that is ARCH=s390x and in the else-branch to BACKEND=s390x ~ z/VM.

File: lib/opensusebasetest.pm (hunk 2 above)
Comment (okurz) [reply lib/opensusebasetest.pm]: You are right. I assumed L400 and the following to be an else-branch which I must not hit but actually it's "elsif (!match_has_tag('grub'))" so I should delete this elsif-branch

File: lib/opensusebasetest.pm (hunk 3 above)
Comment (okurz) [reply lib/opensusebasetest.pm]: yes, I will

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/opensusebasetest.pm]: ah sorry , mess up with indentation :)

File: lib/utils.pm [hunk 5]
```diff
@@ -311,10 +311,17 @@ sub minimal_patch_system {
...
+    workaround_type_encrypted_passphrase()
+
+Record soft-failure for unresolved feature which we think is important and then unlock encrypted boot partitions if we expect it to be encrypted. This condition is met on 'storage-ng' which by default puts the boot partition within the encrypted LVM same as in test scenarios where we explicitly create an LVM including boot (C<FULL_LVM_ENCRYPT>). C<ppc64le> was already doing the same by default also in the case of pre-storage-ng but not anymore for storage-ng.
+
+=cut
 sub workaround_type_encrypted_passphrase {
     if (
         get_var('FULL_LVM_ENCRYPT')
-        || (check_var('ARCH', 'ppc64le')
+        || (   (check_var('ARCH', 'ppc64le') || (get_var('STORAGE_NG') && !check_var('ARCH', 'ppc64le')))
```
Comment (nilxam) [inline lib/utils.pm]: it made cryptlvm test in staging fails to work https://openqa.opensuse.org/tests/603624#step/grub_test/6 as it being true here and try to unlocking.

**Comment by asmorodskyi** [comment]
@okurz can you confirm that someone gives you verbal LGTM at least ? Otherwise it is not how things suppose to happen

**Comment by nicksinger** [comment]
@asmorodskyi he did ask me and I gave him the verbal LGTM. So nobody to blame here

**Comment by nilxam** [comment]
do we have any quick solution here already? or I should add `UNENCRYPTED_BOOT` to cryptlvm test?

**Comment by okurz** [comment]
reverted for now

**Pull request merged**